src/base/static/components/organisms/map-measurement-overlay.tsx

```diff
@@ -0,0 +1,467 @@
+    // relative to the overlay canvas element.
+    const unprojected =
+      webMercatorViewport.current &&
+      webMercatorViewport.current.unproject([evt.x - left, evt.y - top]);
+
+    unprojected && positions.current.push(unprojected);
+    selectedTool.current &&
+      setMeasurementFeatureCollection(
+        buildMeasurementFeatureCollection(
+          selectedTool.current,
+          positions.current,
+        ),
+      );
+  };
+
+  const handleUndoLastPoint = () => {
+    positions.current.pop();
+    selectedTool.current &&
+      setMeasurementFeatureCollection(
+        buildMeasurementFeatureCollection(
+          selectedTool.current,
+          positions.current,
+        ),
+      );
+  };
+
+  const handleReset = () => {
+    positions.current = [];
+    setMeasurementFeatureCollection(featureCollection([]));
+    selectedTool.current = null;
+    setUnits(null);
+    setMeasurement(null);
+    if (overlayRef.current) {
+      overlayRef.current._canvas.style.pointerEvents = "none";
+    }
+  };
+
+  React.useEffect(() => {
+    // Unless I'm missing the right way to use overlays, it seems necessary
+    // to attach a click listener here to capture clicks on the overlay.
+    // See: https://github.com/uber/react-map-gl/issues/470
+    if (overlayRef && overlayRef.current && overlayRef.current._canvas) {
+      overlayRef.current._canvas.addEventListener("click", handleOverlayClick);
+      // This direct manipulation of the `pointer-events` style property feels
+      // like a hack, but the goal is to enable clicks on the underlying map
+      // when no measurement tool is selected.
+      overlayRef.current._canvas.style.pointerEvents = "none";
+    } else {
+      // eslint-disable-next-line no-console
+      console.error(
+        "Measurement overlay: failed to attach click event listener",
+      );
+    }
+
+    return () => {
+      overlayRef &&
+        overlayRef.current &&
+        overlayRef.current._canvas.removeEventListener(
+          "click",
+          handleOverlayClick,
+        );
+    };
+  }, []);
+
+  React.useEffect(
+    () => {
+      webMercatorViewport.current = getNewViewport({
+        ...props.viewport,
+        width: props.width,
+        height: props.height,
+      });
+    },
+    [props.viewport, props.width, props.height],
+  );
+
+  React.useEffect(
+    () => {
+      const {
+        features,
+      }: { features: Feature[] } = measurementFeatureCollection;
+      const measurementFeature = features[features.length - 1] as Feature;
+
+      if (
+        measurementFeature &&
+        measurementFeature.geometry.type == "LineString"
+      ) {
+        setMeasurement(
+          measurementFeature.geometry.coordinates.reduce(
+            (total, nextCoords, i) => {
+              return (
+                i > 0 &&
+                total +
+                  distance(
+                    // @ts-ignore
+                    measurementFeature.geometry.coordinates[i - 1],
```

Comment:
I had to punt typing this... it complains that `coordinates` is not a member of `geometry`, but `measurementFeatureCollection` should be ` Feature` I think, right? I couldn't figure out what was wrong here.

Reply:
Ok, I think I see what was happening here: `geometry` is of type `Geometry`, which itself can be of type `GeometryCollection`, which do not have a `coordinates`. I think the ambiguity of the `measurementFeatureCollection` typing caused this to fail.

Reply:
I'm glad you were able to figure this out :)

I left a few comments above that should clean things up